drivers/xen/dom0/sysctl.c

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2023 EPAM Systems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/arch/arm64/hypercall.h>
+#include <zephyr/xen/dom0/sysctl.h>
+#include <zephyr/xen/generic.h>
+#include <zephyr/xen/public/xen.h>
+
+int xen_sysctl_physinfo(struct xen_sysctl_physinfo *info)
+{
+	xen_sysctl_t sysctl = {0};
+	int ret;
+
+	sysctl.cmd = XEN_SYSCTL_physinfo;
+	ret = do_sysctl(&sysctl);
+	if (ret < 0) {
+		return ret;
+	}
+	*info = sysctl.u.physinfo;
+	return ret;
+}
+
+int xen_sysctl_getdomaininfo(struct xen_domctl_getdomaininfo *domaininfo, int first, int num)
```

Comment:
`int` suggests that `first` and `num` can be negative. This is not true.

Reply:
Done